common/gpu/intel/default.nix

```
@@ -1,26 +1,56 @@
-{ config, lib, pkgs, ... }:
-
 {
-  options.hardware.intelgpu.loadInInitrd = lib.mkEnableOption (lib.mdDoc
-    "loading `i195` kernelModule at stage 1. (Add `i915` to `boot.initrd.kernelModules`)"
-  ) // {
-    default = true;
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+{
+  options.hardware.intelgpu.driver = lib.mkOption {
+    description = "Intel GPU driver to use";
+    type = lib.types.enum [
+      "i915"
+      "xe"
+    ];
+    default = "i915";
   };
 
-  config = lib.mkMerge [
-    (lib.mkIf config.hardware.intelgpu.loadInInitrd { 
-      boot.initrd.kernelModules = [ "i915" ]; 
-    })
-    {
-      environment.variables = {
-        VDPAU_DRIVER = lib.mkIf config.hardware.opengl.enable (lib.mkDefault "va_gl");
-      };
+  options.hardware.intelgpu.loadInInitrd =
+    lib.mkEnableOption (
+      lib.mdDoc "Load the Intel GPU kernel module at stage 1 boot. (Added to `boot.initrd.kernelModules`)"
+    )
+    // {
+      default = true;
+    };
+
+  config = {
+    boot.initrd.kernelModules = [ config.hardware.intelgpu.driver ];
 
-      hardware.opengl.extraPackages = with pkgs; [
-        (if (lib.versionOlder (lib.versions.majorMinor lib.version) "23.11") then vaapiIntel else intel-vaapi-driver)
-        libvdpau-va-gl
-        intel-media-driver
-      ];
-    }
-  ];
+    environment.variables = {
+      VDPAU_DRIVER = lib.mkIf config.hardware.opengl.enable (lib.mkDefault "va_gl");
+    };
+
+    hardware.opengl.extraPackages = with pkgs; [
+      (
+        if (lib.versionOlder (lib.versions.majorMinor lib.version) "23.11") then
+          vaapiIntel
+        else
+          intel-vaapi-driver
+      )
+      libvdpau-va-gl
+      intel-media-driver
+    ];
+
+    assertions = [
+      {
+        assertion = (
+          config.hardware.intelgpu.driver == "i915"
+          || (
+            config.hardware.intelgpu.driver == "xe"
+            && lib.versionAtLeast config.boot.kernelPackages.kernel.version "6.8"
+          )
```

Minor simplification nit:

```suggestion
          config.hardware.intelgpu.driver != "xe"
          || lib.versionAtLeast config.boot.kernelPackages.kernel.version "6.8"
```